src/scene/morph-instance.js

```diff
@@ -170,25 +174,47 @@ class MorphInstance {
         return clone;
     }
 
+    _getWeightIndex(key) {
+        const index = this._weightMap.get(key);
+        if (index) {
+            return this._weights[index];
+        }
+        Debug.error('Attempting to retrieve a morph target weight using a key that doesn\'t exist.');
+    }
+
     /**
      * Gets current weight of the specified morph target.
      *
-     * @param {number} index - An index of morph target.
+     * @param {string|number} key - An identifier for the morph target. Either the weight index or the weight name
      * @returns {number} Weight.
      */
-    getWeight(index) {
+    getWeight(key) {
+        let index;
+        if (typeof key === 'string') {
+            this._getWeightIndex(key);
+        } else {
+            index = key;
+        }
         return this._weights[index];
     }
 
     /**
      * Sets weight of the specified morph target.
      *
-     * @param {number} index - An index of morph target.
+     * @param {string|number} key - An identifier for the morph target. Either the weight index or the weight name
      * @param {number} weight - Weight.
      */
-    setWeight(index, weight) {
-        this._weights[index] = weight;
-        this._dirty = true;
+    setWeight(key, weight) {
+        let index;
+        if (typeof key === 'string') {
+            index = this._getWeightIndex(key);
+        } else {
+            index = key;
+        }
+        if (Number.isFinite(this._weights[key])) {
```

Comment:
Sorry this was a bug from the refactor. Weights should be retrieved using the index instead of the key. Then if we cant retrieve a number from the weights array, we shouldn't be setting a new value or flagging the MorphInstance as dirty.